app/scripts/migrations/097.ts

```diff
@@ -0,0 +1,46 @@
+import { isNullOrUndefined } from '@metamask/utils';
+import { cloneDeep } from 'lodash';
+
+type VersionedData = {
+  meta: { version: number };
+  data: Record<string, unknown>;
+};
+
+export const version = 97;
+
+/**
+ * This migration does the following:
+ *
+ * - Deletes currently stored advancedGasFee in preferences controller, replacing the default with an empty object
+ *
+ * @param originalVersionedData - Versioned MetaMask extension state, exactly what we persist to dist.
+ * @param originalVersionedData.meta - State metadata.
+ * @param originalVersionedData.meta.version - The current state version.
+ * @param originalVersionedData.data - The persisted MetaMask state, keyed by controller.
+ * @returns Updated versioned MetaMask extension state.
+ */
+export async function migrate(
+  originalVersionedData: VersionedData,
+): Promise<VersionedData> {
+  const versionedData = cloneDeep(originalVersionedData);
+  versionedData.meta.version = version;
+  migrateData(versionedData.data);
+  return versionedData;
+}
+
+function migrateData(state: Record<string, unknown>): void {
+  changeShapeAndRemoveOldAdvancedGasFeePreference(state);
+}
+
+function changeShapeAndRemoveOldAdvancedGasFeePreference(
+  state: Record<string, any>,
+) {
+  if (isNullOrUndefined(state.PreferencesController)) {
+    return;
+  }
+
+  state.PreferencesController = {
+    ...(state.PreferencesController ?? {}),
+    advancedGasFee: {},
```

Comment:
If we erase the `advancedGasFee` property completely, we lose information about whether the user had set this to something before. We may want to alert users who had used this feature that we're resetting it.

Perhaps we could set a flag somewhere (maybe in the app state controller?) to show whether a user had their advanced gas fee setting erased? Then we could use that to trigger a Whats New announcement

Reply:
I also asked for copy for the whats new announcement to be added later.